zulipterminal/config/themes.py

```diff
@@ -540,4 +541,70 @@ def theme_with_monochrome_added(theme: ThemeSpec) -> ThemeSpec:
         else:  # 1-to-1 mapping (same as other style)
             new_style = style
         updated_theme.append(new_style)
-    return updated_theme
+    return theme
+
+
+def add_color_prop(colors, *prop):
+    """
+    Adds properties(Bold, Italics, etc...) to Enum Colors
+    Use: add_color_prop(Color, 'BOLD', 'ITALICS', 'STRIKETHROUGH')
+    """
+    prop_n = "_".join([p.upper() for p in prop])
+    prop_v = " , ".join([p.lower() for p in prop])
+    colors = Enum(
+        "Color",
+        {
+            **{c.name: c.value for c in colors},
+            **{c.name + f"__{prop_n}": c.value + f" , {prop_v}" for c in colors},
+        },
+    )
+    return colors
+
+
+from zulipterminal.config import gruvbox
+
+ZT_THEMES = {"gruvbox_dark": gruvbox}
+
+
+def generate_theme(theme_name, color_depth):
+    theme_styles = ZT_THEMES[theme_name].STYLES
+
+    urwid_theme = []
+    for style_name, style_conf in theme_styles.items():
+        fg, bg = style_conf
+        fg_codes = fg.value.split()
+        bg_codes = bg.value.split()
+
+        if color_depth == 1:
+            # FIXME: Check for completeness of required_styles
+            try:
+                new_style = (style_name, "", "", required_styles[style_name])
+            except:
+                pass
+        elif color_depth == 16:
+            fg_name = fg.name.lower().split("__")[0].replace("_", " ")
+            bg_name = bg.name.lower().split("__")[0].replace("_", " ")
+            fg = fg_name + " ".join(fg_codes[2:])
+            bg = bg_name + " ".join(bg_codes[2:])
+            new_style = (style_name, fg, bg)
+        elif color_depth == 256:
+            fg = " ".join(fg_codes[:1] + fg_codes[2:]).replace("_", " ")
+            bg = " ".join(bg_codes[:1] + bg_codes[2:]).replace("_", " ")
+            new_style = (style_name, "", "", "", fg, bg)
+        elif color_depth == 2 ** 24:
+            fg = " ".join(fg_codes[1:]).replace("_", " ")
+            bg = " ".join(bg_codes[1:]).replace("_", " ")
+            new_style = (style_name, "", "", "", fg, bg)
+        urwid_theme.append(new_style)
+        print(new_style)
+    return urwid_theme
+
+
+if __name__ == "__main__":
+    generate_theme("gruvbox_dark", 1)
```

Comment:
fixing `msg__bold` to `msg_bold` seems to have fixed this

Reply:
I assume it was some kind of automatic processing issue?

Reply:
Yes I would think so.

Reply:
Yes this happened because I used `pass` instead of `continue` for mono theme. I've fixed it.
The try statement is also removed after the REQUIRED_STYLES update commit.